src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -2324,194 +2324,195 @@ private void filterExistingFiles() throws SnowflakeSQLException {
 
       logger.debug("received object summaries from remote storage");
 
-      Iterator<StorageObjectSummary> objectSummariesIterator;
```

Comment:
The only two changes in this file are 1) replace explicit calling of iterator apis with for-each loop; 2) wrap the for-each loop with a do-while loop and try-catch blocks for retry purpose.

The main body logic in the for-each loop (original while-true loop) is not changed.

Reply:
One suggestion: is it possible to divide the changes into two PRs. One is just refactoring the current code to make it methods and the second one add the loop changes. I know the original coding style is not optimal so we could improve this time.

Reply:
I see. Nice suggestion!